prov/shm/src/smr_ep.c

```diff
@@ -298,6 +298,80 @@ void smr_format_iov(struct smr_cmd *cmd, const struct iovec *iov, size_t count,
 	memcpy(cmd->msg.data.iov, iov, sizeof(*iov) * count);
 }
 
+int smr_format_mmap(struct smr_ep *ep, struct smr_cmd *cmd,
+		    const struct iovec *iov, size_t count, size_t total_len,
+		    struct smr_tx_entry *pend, struct smr_resp *resp)
+{
+	char shm_name[NAME_MAX];
+	void *mapped_ptr;
+	int fd, ret, num;
+	uint64_t msg_id;
+	struct smr_ep_name *map_name;
+
+	msg_id = ep->msg_id++;
+	num = mmap_shm_name(shm_name, ep->name, msg_id);
+	if (num < 0) {
+		FI_WARN(&smr_prov, FI_LOG_AV, "generating shm file name failed\n");
+		return -errno;
+	}
+
+	fd = shm_open(shm_name, O_RDWR | O_CREAT, S_IRUSR | S_IWUSR);
+	if (fd < 0) {
+		FI_WARN(&smr_prov, FI_LOG_EP_CTRL, "shm_open error\n");
+		return -errno;
+	}
+
+	map_name = calloc(1, sizeof(*map_name));
+	if (!map_name) {
+		FI_WARN(&smr_prov, FI_LOG_EP_CTRL, "calloc error\n");
+		ret = -FI_ENOMEM;
+		goto unlink_close;
+	}
+	strncpy((map_name)->name, shm_name, NAME_MAX - 1);
```

Comment:
We can save a copy and simplify the error case for this is we allocate map_name first and then call mmap_shm_name with map_name->name

Reply:
Yes, thank you. Updated.